src/Microsoft.VisualStudio.Editors/PropPages/CodeAnalysisPropPage.vb

```diff
@@ -19,20 +21,45 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
             'Add any initialization after the InitializeComponent() call
             AddChangeHandlers()
+
+            PopulateAnalysisLevelComboBoxItems()
         End Sub
 
         Protected Overrides ReadOnly Property ControlData As PropertyControlData()
             Get
                 If m_ControlData Is Nothing Then
                     m_ControlData = New PropertyControlData() {
                         New PropertyControlData(1, "RunAnalyzersDuringBuild", RunAnalyzersDuringBuild, ControlDataFlags.None),
-                        New PropertyControlData(2, "RunAnalyzersDuringLiveAnalysis", RunAnalyzersDuringLiveAnalysis, ControlDataFlags.PersistedInProjectUserFile)
+                        New PropertyControlData(2, "RunAnalyzersDuringLiveAnalysis", RunAnalyzersDuringLiveAnalysis, ControlDataFlags.PersistedInProjectUserFile),
+                        New PropertyControlData(3, "EnableNETAnalyzers", EnableNETAnalyzersCheckBox, ControlDataFlags.None),
+                        New PropertyControlData(4, "AnalysisLevel", AnalysisLevelComboBox, AddressOf AnalysisLevelSet, AddressOf AnalysisLevelGet, ControlDataFlags.None)
```

Comment:
Should this control only be enabled when "EnabledNETAnalyzers" is true?

Reply:
I thought about it and decided to not disable the control (and set its value to `none`) since there is a case where you would want to configure both.

So `AnalysisLevel` set to "none"  implicitly disables analyzers (and therefore automatically unchecks the `EnableNETAnalyzersCheckBox`)

However it is valid to have `<EnableNETAnalyzers>false</EnableNETAnalyzers>` with `<AnalysisLevel>5.0</AnalysisLevel>`. This indicates that you do not want any analyzers to give you warnings about mis-using apis but you still want new language warnings from the compiler.

Maybe this is too much of a corner case and we should just tie the two together?

Reply:
Unclear to me at this point. My question was motivated by thinking how this might work in the new property pages. If you feel there's a better experience here, let's work through a design.

The UI rarely offers the full range of configuration options that exist by directly editing the project file. If we can simplify and improve the experience for 99% of users, I'm for it.